experimenter/experimenter/nimbus_ui_new/templates/nimbus_experiments/detail.html

```diff
@@ -393,6 +395,33 @@ <h6>
           </tbody>
         </table>
       </div>
+      <div class="modal fade"
+           id="promoteToRolloutModal-{{ branch.slug }}"
+           tabindex="-1"
+           aria-labelledby="promoteToRolloutModalLabel-{{ branch.slug }}"
+           aria-hidden="true">
+        <div class="modal-dialog">
+          <div class="modal-content">
+            <div class="modal-header">
+              <h1 class="modal-title fs-5">Promote {{ branch.slug }} to Rollout</h1>
+              <button type="button"
+                      class="btn-close"
+                      data-bs-dismiss="modal"
+                      aria-label="Close"></button>
+            </div>
+            <div class="modal-body bg-body-tertiary">
+              <form id="promoteToRolloutForm-{{ branch.slug }}"
+                    hx-post="{% url 'nimbus-new-promote-to-rollout' experiment.slug branch.slug %}"
```

Comment:
Oh I see, you put them in the url, and then unpack them and pass them straight to the form.  Yeah that totally works.  I think you could also make them actual form fields and then use [`hx-vals`](https://htmx.org/attributes/hx-vals/) to pass them along, so that way everything is just automatically passed in to the form through `data` and you don't need to explicitly unpack them and pass them in.

I wouldn't say it's strictly necessary, just might remove some special cases.  I think you could keep the experiment slug in the url so that the object gets looked up and passed as `instance`, and pass the branch slug as a form field, and that way both are looked up automatically and passed along.

Just a thought, not strictly necessary.